contracts/DecentHats_0_1_0.sol

```diff
@@ -113,7 +98,47 @@ contract DecentHats_0_1_0 {
             );
     }
 
-    function createHatAccountMint(
+    function createAccount(
+        IERC6551Registry _registry,
+        address _hatsAccountImplementation,
+        bytes32 salt,
+        address protocolAddress,
+        uint256 hatId
+    ) internal returns (address) {
+        return
+            _registry.createAccount(
+                _hatsAccountImplementation,
+                salt,
+                block.chainid,
+                protocolAddress,
+                hatId
+            );
+    }
+
+    function createTopHatAndAccount(
+        IHats _hatsProtocol,
+        string memory _topHatDetails,
+        string memory _topHatImageURI,
+        IERC6551Registry _registry,
+        address _hatsAccountImplementation,
+        bytes32 salt
+    ) internal returns (uint256 topHatId, address topHatAccount) {
+        topHatId = _hatsProtocol.mintTopHat(
+            address(this),
+            _topHatDetails,
+            _topHatImageURI
+        );
+
+        topHatAccount = createAccount(
+            _registry,
+            _hatsAccountImplementation,
+            salt,
+            address(_hatsProtocol),
+            topHatId
+        );
+    }
+
+    function createHatAndAccountAndMintAndStreams(
```

Comment:
Can we rename it to `doTheWholeThingBuddy`?

Reply:
Joke ofc 🤣

Reply:
lol yeah i know, this function name is kind of insane